File: backend/EventifyApi/Models/Entities/Location.cs
namespace EventifyApi.Models.Entities;




public class Location
{
    
    
    
    public int Id { get; set; }

    
    
    
    public string Name { get; set; } = string.Empty;

    
    
    
    public string Address { get; set; } = string.Empty;

    
    
    
    public int Capacity { get; set; }

    
    
    
    public string? Description { get; set; }

    
    
    
    public string? ImageUrl { get; set; }

    
    
    
    public bool IsActive { get; set; } = true;

    
    
    
    public double? Latitude { get; set; }

    
    
    
    public double? Longitude { get; set; }

    
    
    
    public string? ContactEmail { get; set; }

    
    
    
    public string? ContactPhone { get; set; }

    
    
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    
    
    
    public DateTime? UpdatedAt { get; set; }

    
    
    

    
    
    
    public ICollection<Event> Events { get; set; } = new List<Event>();
}

File: frontend/src/types/location.ts







export interface Location {
  
  id: number
  
  
  name: string
  
  
  address: string
  
  
  capacity: number
  
  
  description?: string
  
  
  imageUrl?: string
  
  
  isActive: boolean
  
  
  createdAt: string
  
  
  updatedAt?: string
  
  
  latitude?: number
  longitude?: number
  
  
  contactEmail?: string
  contactPhone?: string
  
  
  amenities?: string[]
}


export interface LocationSummary {
  
  id: number
  
  
  name: string
  
  
  address: string
  
  
  capacity: number
  
  
  imageUrl?: string
  
  
  isActive: boolean
}






export interface CreateLocationDto {
  
  name: string
  
  
  address: string
  
  
  capacity: number
  
  
  description?: string
  
  
  imageUrl?: string
  
  
  latitude?: number
  longitude?: number
  
  
  contactEmail?: string
  contactPhone?: string
  
  
  amenities?: string[]
}


export interface UpdateLocationDto {
  
  name?: string
  
  
  address?: string
  
  
  capacity?: number
  
  
  description?: string
  
  
  imageUrl?: string
  
  
  isActive?: boolean
  
  
  latitude?: number
  longitude?: number
  
  
  contactEmail?: string
  contactPhone?: string
  
  
  amenities?: string[]
}






export interface LocationSearchParams {
  
  search?: string
  
  
  isActive?: boolean
  
  
  minCapacity?: number
  
  
  maxCapacity?: number
  
  
  sortBy?: 'name' | 'capacity' | 'createdAt'
  
  
  sortOrder?: 'asc' | 'desc'
  
  
  page?: number
  
  
  pageSize?: number
}






export interface LocationOption {
  
  value: number
  
  
  title: string
  
  
  subtitle: string
  
  
  capacity: number
  
  
  disabled: boolean
}
